src/scratch.h

```diff
@@ -0,0 +1,35 @@
+/**********************************************************************
+ * Copyright (c) 2017 Andrew Poelstra	                              *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef _SECP256K1_SCRATCH_
+#define _SECP256K1_SCRATCH_
+#include <sys/types.h>
+
+/* The typedef is used internally; the struct name is used in the public API
+ * (where it is exposed as a different typedef) */
```

Comment:
The typedef and struct name are the same publicly; the struct is just not defined there.

Reply:
I'm not sure what you mean, the typedef here is `secp256k1_scratch`, the public one is `secp256k1_scratch_space`.

Reply:
Oops, ignore me!